src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -763,9 +763,9 @@ private static string GetMatchingFrameworkStrings(PackageSpec spec, NuGetFramewo
         private static XElement GeneratePackagePathProperty(LocalPackageInfo localPackageInfo)
         {
 #if NETCOREAPP
-            return GenerateProperty($"Pkg{localPackageInfo.Id.Replace(".", "_", StringComparison.Ordinal)}", localPackageInfo.ExpandedPath);
+            return GenerateProperty($"Pkg{localPackageInfo.Id.Replace(".", "_", StringComparison.Ordinal)}", $"{localPackageInfo.ExpandedPath}{Path.DirectorySeparatorChar}");
 #else
-            return GenerateProperty($"Pkg{localPackageInfo.Id.Replace(".", "_")}", localPackageInfo.ExpandedPath);
+            return GenerateProperty($"Pkg{localPackageInfo.Id.Replace(".", "_")}", $"{localPackageInfo.ExpandedPath}{Path.DirectorySeparatorChar}");
```

Comment:
String concatenation is faster than `string.Format`. Source: https://www.meziantou.net/performance-string-concatenation-vs-string-format-vs-interpolated-string.htm. Also the book "Writing High-Performance .NET Code" by Ben Watson, if you haven't read it and it's the type of thing that interests you.

*([interprolated strings are just syntatic sugar for `string.Format`](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQyAMABMugHQCSA8gNw47IDMRqBAwgQN44HdGPGEsA9gDsAxgENgACgDKwAE5hhAcwJKADgFdgASg5ceh5AHYCAEgBE7DdoC+7AAqSAFiQAiYeQFNRwQfIBPGS91cXlJfxZnMNsLGmxDWxxbIA=))

Reply:
Yes string concatenation is faster, I don't know if it matters in this case.  The allocation is the same and the speed difference is negligible since this isn't a very hot path.  This method also returns the value from `GenerateProperty()` which itself uses string interpolation.  

I don't have a strong opinion honestly, I do lean towards readability and consistency.  I can use string concatenation but I would want to update this class to use it everywhere.  What do you think?

Reply:
I don't understand how `$"{localPackageInfo.ExpandedPath}{Path.DirectorySeparatorChar}"` is more (or less) readable than `localPackageInfo.ExpandedPath + Path.DirectorySeparatorChar`. To me they're both the same, like different regional accents or expressions, except this is software development where one of them has worse runtime performance. While it's not on a hot path, I'd like to avoid "death by one thousand cuts". If there are trivial improvements, why not make them?

If you want to make it both more readable and debuggable, split it into separate lines, to make it clear that there are multiple expressions going on. Code golf is not a good thing.

```cs
string propertyName = "Pkg" + localPackageInfo.Id.Replase(".", "_");
string propertyValue = localPackageInfo.ExpandedPath + Path.DirectorySeparatorChar;
string result = GenerateProperty(propertyName, propertyValue);
return result;
```

Reply:
> If there are trivial improvements, why not make them?

The same could be said about making trivial improvements at the sacrifice of code maintainability, readability, regressions, etc.  I generally lean towards string interpolation as a best practice when making strings from other parts.  This line of code already has string interpolation as it is.  I'm just editing the code to add a feature, it seems reasonable to follow the pattern established in the file.  Like I said, I have no strong opinion in this particular case, I like your suggestion of updating this method to use concatenation, do you think its worth updating this whole file as well?  Or should I just walk away and we can clean it up in a subsequent perf effort?

Reply:
I've pushed two commits, one uses concatenation for this method and the other uses concatenation for the whole file.  Using concatenation in the whole file would probably show a measurable improvement, I'm just hoping tests have good coverage in this area.

Reply:
> The same could be said about making trivial improvements at the sacrifice of code maintainability, readability, regressions, etc

😕 ummm, what? I feel like there's some stawman arguments going on here. I said when there are two equivalent syntaxes, both of which are just as easy to understand as each other, choose the one with better performance. I don't understand how you jumped to "sacrifice code maintainability, readability, regressions".

All of this because of `$"{arg1}{arg2}"` vs `arg1 + arg2`. Extending my suggestion to anything other than concatenation of two strings is a misunderstanding of what I was trying to communicate. The string string concat has fewer characters compared to the interpolated string version. The interoplated string version has zero whitespace characters, whereas the string concat version has two whitespace characters, which I'd argue makes it easier to read since anyone growing up in a country that speaks a language that uses spaces as word separators are used to a lifetime of reading which trains our brains to more easily process spaces than `$"{}` characters.

> I generally lean towards string interpolation as a best practice when making strings from other parts.

"I generally" is effectively an agreement that it's a personal preference, not an objective fact, and you won't be surprised to hear that I disagree. If the team agrees to prefer interpolated strings, then fine I'll adapt, but we have no such team coding guideline that I'm aware of. In the last two years we've spent a lot of effort in improving performance, even looking for small wins, so when we can make these small wins with no harm to readability, maintaibility, regressions, I'll take/suggest them.

> do you think its worth updating this whole file as well?

I'm fine with only changing the methods that will be modified anyway as part of the feature/bug fix. I've had disagreements with other people in the team in the past who feel more strongly that a file or project should consistently apply the same pattern thoughout, and another who thinks that anything, no matter how minor, that is not directly related to the issue assigned, is out of scope and should be done in a different issue/PR. I argue that this raises the bar too high to make improvements in scenarios where we all agree the pattern is not ideal, especially when one of the people who made this argument also doesn't like PRs that change too much code unrelated to the feature/bug fix. As someone who has spent (almost) my entire career since finishing university working on applications that were ~10 years old before I joined the team, I'm heavily biased in my experiences of maintaining "legacy" code and trying to make improvements to the codebase over time since we never ever get time to re-write from scratch or make non-functional refactorings. Basically, I like and try to follow the "boy scout rule" as described in the Clean Code book.